File: second largest.py
class Solution:
    def getSecondLargest(self, arr):
        if len(arr) < 2:
            return -1

        first = second = -1

        for num in arr:
            if num > first:
                second = first
                first = num
            elif num > second and num != first:
                second = num

        return second if second != -1 else -1

File: second largest.c
int getSecondLargest(int *arr, int n) {
    if (n < 2) return -1;

    int first = -1, second = -1;

    for (int i = 0; i < n; i++) {
        if (arr[i] > first) {
            second = first;
            first = arr[i];
        } else if (arr[i] > second && arr[i] != first) {
            second = arr[i];
        }
    }

    return (second == -1) ? -1 : second;
}
